File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairToIndex = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int remaining = target - nums[i];
            if (pairToIndex.containsKey(remaining)) {
                return new int[]{pairToIndex.get(remaining), i};
            }
            pairToIndex.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        n = len(nums)
        i, j = 0, n-1

        while i <= j:
            mid = ( i + j ) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                j = mid - 1
            elif nums[mid] < target:
                i = mid + 1

        return -1
        